packages/backend/src/apps/delay/actions/delay-until/index.ts

```diff
@@ -11,16 +13,42 @@ export default defineAction({
       key: 'delayUntil',
       type: 'string' as const,
       required: true,
-      description: 'Delay until the date. E.g. 2022-12-18',
+      description: 'Delay until the date. E.g. 25 Aug 2023',
+      variables: true,
+    },
+    {
+      label: 'Delay until (Time)',
+      key: 'delayUntilTime',
+      type: 'string' as const,
+      required: false,
+      description: 'Delay until the time (24h). E.g. 08:00, 23:00',
       variables: true,
     },
   ],
 
   async run($) {
-    const { delayUntil } = $.step.parameters
+    const defaultTime = '00:00'
+    const { delayUntil, delayUntilTime } = $.step.parameters
+    const delayTimestamp = generateTimestamp(
+      delayUntil as string,
+      (delayUntilTime as string) === ''
+        ? defaultTime
+        : (delayUntilTime as string),
+    )
+
+    if (isNaN(delayTimestamp)) {
+      throw new Error(
+        'Invalid timestamp entered, please check that you keyed in the date and time in the correct format',
+      )
+    }
+
+    if (!delayUntilTime) {
+      $.step.parameters.delayUntilTime = defaultTime // allow users to use it as a variable even though it is optional
```

Comment:
nit: This comment might be slightly misleading, `parameters` doesn't have anything to do with output variables

We might instead want to do this at the bottom:

```
const dataItem = {
  delayUntil,
  delayUntilTime: delayUntilTime ?? defaultTime,
}
```